src/ArrayUtils.f90

```diff
@@ -991,4 +995,69 @@ PURE FUNCTION findUpBound_1DReal(r,pos,delta,incl,tol) RESULT(val)
   ENDIF
 ENDFUNCTION findUpBound_1DReal
 !
+!-------------------------------------------------------------------------------
+!> @brief Routine that checks that the entries of r either monotonically increase or decrease.
+!>        Note this is limited to 1 dimensional real arrays.
+!> @param r input array to check for monotonicity of values
+!> @returns good boolean indicating whether the check has passed or not
+!>
+FUNCTION Is_Mono(r) RESULT(good)
+  REAL(SRK),INTENT(IN) :: r(:)
+  LOGICAL(SBK) :: good
+
+  IF(SIZE(r) > 1) THEN
+    IF(r(1) < r(2)) THEN
+      good=Is_Mono_Increasing(r)
+    ELSE
+      good=Is_Mono_Decreasing(r)
+    ENDIF
+  ELSE
+    good=.FALSE.
```

Comment:
should set this above, no else block

Reply:
Guilty until proven innocent. I like it.